### PR TITLE
Update webpack-related dependencies

### PR DESCRIPTION
**Pull request by ChumpChief**
Older webpack versions are getting flagged by dependabot (CVE-2024-43788).  This PR would update those along with a variety of webpack-adjacent dependencies (webpack-dev-server, loaders, plugins, etc.).  I reviewed the release notes for each of these that crossed a major version boundary and didn't see anything obviously problematic.  I've manually tested all the affected examples, found no regressions (and filed bugs for the preexisting breaks).

Aside from the simple updates, a few items of note:
1. The docs assets bundle looks substantially different with new webpack.  Tested manually with local docs deployment and confirmed everything still works the same as before.
    * Side note: Ironically I also wasn't able to successfully build the bundle without updating the dependencies, as I hit an error `Error: error:0308010C:digital envelope routines::unsupported`.  Building with the updated dependencies works reliably.
1. clean-webpack-plugin is obsolete as of [webpack 5.20](https://webpack.js.org/configuration/output/#outputclean), removed remaining references to it and adjusted as needed.
1. Extended the test timeout for server lambda's webpack as a precaution - it was usually finishing within the timeout but I was getting an occasional timeout locally.

### CODE REVIEW

**Comment by ChumpChief** [comment]
webpack-dev-server is being weird about esm/cjs so I might back that down to 4.15.2 for now and try again later.  My local incremental build had skipped over webpack-fluid-loader for some reason (not sure why) so I didn't see the error locally at first :-/

**Comment by Josmithr** [comment]
> Side note: I found the instructions in the README to not work as written (error on yarn install: Usage Error: This project is configured to use pnpm), but found success after pulling the "package" out to its own directory (such that pnpm wouldn't get found).

I'm surprised to hear that the README has such guidance. The docs dir uses pnpm like everything else. It never used yarn to my knowledge. We should definitely fix this.

**Comment by tylerbutler** [comment]
> > Side note: I found the instructions in the README to not work as written (error on yarn install: Usage Error: This project is configured to use pnpm), but found success after pulling the "package" out to its own directory (such that pnpm wouldn't get found).
> 
> ~I'm surprised to hear that the README has such guidance. The docs dir uses pnpm like everything else. It never used yarn to my knowledge. We should definitely fix this.~
> 
> Never mind. I see this is regarding the theme code. I have no knowledge about that :)

This was inherited from the vscode team, and in general, we should not touch the bundled code unless absolutely necessary, or with a lot of testing of the site. The bundled code does stuff related to nav as I recall.

File: docs/themes/thxvscode/assets/js/package.json [hunk 1]
```diff
@@ -3,14 +3,14 @@
 		"build": "webpack --mode=production"
 	},
 	"devDependencies": {
```
Comment (tylerbutler) [inline docs/themes/thxvscode/assets/js/package.json:5]: Add `"packageManager": "yarn@1.22.22"` to this file and yarn should work as expected.

Comment (ChumpChief) [reply docs/themes/thxvscode/assets/js/package.json:5]: That did it!  Added in latest.

**Comment by ChumpChief** [comment]
> This was inherited from the vscode team, and in general, we should not touch the bundled code unless absolutely necessary, or with a lot of testing of the site. The bundled code does stuff related to nav as I recall.

On the fortunate side, it doesn't do much so I was able to manually test the full set of functionality.
* accessibility.js tries to make a "skip to content" button work, but AFAICT we don't actually have such a button anywhere on our site.  If we did, I think in most places its logic would get tripped up on our breadcrumb nav and try to scroll that into view.
* docs.js does a few things:
    * Uses scrollspy to highlight the current section in the right nav - continues to work
    * Sets focus on the narrow-width layout's nav dropdown based on localStorage - doesn't seem to do much, but continues to work
    * Sets a bootstrap affix on the right nav - doesn't really seem to have an affect?  There is affix information in the html, so I think this is duplicative?  In any case, the affix works and the code here runs.
    * Uses stickyfill for the navbar container - would work, but doesn't because its wrapping element is zero-height.  Also somewhat unnecessary since position:sticky is supported by >99% of browsers now.
    * Some nav expand/collapse code (primarily for the left nav) - still works
    * More narrow-width layout nav dropdown code, just to make it navigate.  Still works.
    * Some questionable userAgent detection that fortunately doesn't seem to do anything
    * Some logic to make the hash-link (for article sections) appear on hover.  Still works.

Although there are other files in the directory, those two are the only ones that make it into the bundle via index.js.  Most of it could probably be simplified to remove dependency on the alternate build/packageManager, and/or the dependency on jQuery/Bootstrap, but it all still seems to work fine at least.

**Comment by ChumpChief** [comment]
Completed my test pass over the examples - found lots of breakages but they all also repro in main so I think this PR is feeling pretty good!  Will open work items for those and report back here.

**Comment by ChumpChief** [comment]
In total, opened 4 bugs:
* AB#14652 - prosemirror breaks on load
* AB#14942 - attributable-map breaks when run
* AB#14943 - memory-orderer has non-functioning test:webpack script
* AB#14944 - local-driver generates an ever-growing number of scribeDeltas entries and quickly blows its quota

**Review by tylerbutler — APPROVED** [review]
Since you took the time to write out everything the bundled code does, could you add it to the readme? Also review the PR description because parts of it are outdated now. Otherwise looks good.

**Comment by ChumpChief** [comment]
> Since you took the time to write out everything the bundled code does, could you add it to the readme? Also review the PR description because parts of it are outdated now. Otherwise looks good.

Did one better and commented it directly inline in the code in latest commit :). Will update the PR description too.

**Pull request merged**